core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -131,8 +137,8 @@ private void updateComputer(Node n, Map<String,Computer> byNameMap, Set<Computer
                     }
                 }
             }
+            used.add(c);
```

Comment:
Nothing is documented about `Node#createComputer()` . The core itself does not operate with nulls from what I see, but there is a case for that :( https://github.com/jenkinsci/gcloud-plugin/blob/4b84dbdd2532dc65feb5bef11f82f713360622b2/src/main/java/com/cloudbees/jenkins/plugins/gcloud/GoogleSlave.java#L21